### PR TITLE
fby3: vf: Add plat_m2 & plat_power_seq &plat_hwmon

### PR DESCRIPTION
**Pull request by HungYi-Li**
Summary:
- Add plat_hwmon.c
- Add plat_m2
- Add plat_power_seq
- Modify sensor table

Test plan:
- Build code : This will be tested after a few commit.

### CODE REVIEW

File: meta-facebook/yv3-vf/src/platform/plat_hwmon.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	msg.rx_len = sizeof(buf);
+	memcpy(&msg.data[0], &reg, sizeof(reg));
+
+	if (i2c_master_read(&msg, retry)) {
+		printf("MB CPLD read failed!\n");
+		return false;
+	}
+
+	buf = msg.data[0];
+	buf = (buf & ~(1 << prsnt_ofs)) | (!val << prsnt_ofs);
```
Comment (garnermic) [inline meta-facebook/yv3-vf/src/platform/plat_hwmon.c:61]: Can you address the linter issue on line 61?

File: meta-facebook/yv3-vf/src/platform/plat_power_seq.c [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+		clkbuf_oe_en(idx, en_clk);
+		delay_function(3, fm_p12v_sw_en, idx, en_12v);
+	}
+
+	if (force_ctl_3v3 && (enable == 0)) {
+		is_force_disable_3v3[idx] = 1;
+		fm_p3v3_sw_en(idx, en_3v3);
+	} else if (force_ctl_3v3 && (enable == 1)) {
+		is_force_disable_3v3[idx] = 0;
+	}
```
Comment (garnermic) [inline meta-facebook/yv3-vf/src/platform/plat_power_seq.c:200]: Please add an else statement to catch any fall through or error conditions.

File: meta-facebook/yv3-vf/src/platform/plat_power_seq.c [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+}
+
+uint8_t m2_dev_power_switch(uint8_t idx, uint8_t enable)
+{
+	return m2_dev_power_switch_with_pwrdis_chk(idx, enable, 1, 0);
+}
+
+uint8_t device_all_power_set(uint8_t idx, uint8_t set_val)
+{
+	uint8_t is_on = (set_val & DEV_PWR_ON ? 1 : 0);
```
Comment (garnermic) [inline meta-facebook/yv3-vf/src/platform/plat_power_seq.c:227]: Please address the linter comments in lines 227, 228 and 229.

**Review by garnermic — CHANGES_REQUESTED** [review]
Minor changes requested.

File: meta-facebook/yv3-vf/src/platform/plat_hwmon.c (hunk 1 above)
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/src/platform/plat_hwmon.c:61]: reformat on #469

File: meta-facebook/yv3-vf/src/platform/plat_power_seq.c (hunk 2 above)
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/src/platform/plat_power_seq.c:200]: reformat on #469

File: meta-facebook/yv3-vf/src/platform/plat_power_seq.c (hunk 3 above)
Comment (HungYi-Li) [reply meta-facebook/yv3-vf/src/platform/plat_power_seq.c:227]: fix it on #469

**Pull request closed without merging**